src/module.c

```diff
@@ -10987,10 +11123,429 @@ sds genModulesInfoString(sds info) {
+    return module_config->get_fn.get_numeric(module_config->name, module_config->privdata);
+}
+
+/* This function takes a module and a list of configs stored as sds NAME VALUE pairs.
+ * It attempts to call set on each of these configs. */
+int loadModuleConfigs(RedisModule *module) {
+    listIter li;
+    listIter li_supplied;
+    listNode *ln;
+    listNode *ln_supplied;
+    int count = 0;
+    sds *config_pair;
+    const char *err = NULL;
+
+    listRewind(module->module_configs, &li);
+    while ((ln = listNext(&li))) {
+        ModuleConfig *module_config = ln->value;
+        int config_supplied = 0;
+        sds config_name = sdscatfmt(sdsempty(), "%s.%s", module->name, module_config->name);
+        listRewind(server.module_configs_queue, &li_supplied);
+        while ((ln_supplied = listNext(&li_supplied))) {
+            sds supplied_config = ln_supplied->value;
+            config_pair = sdssplitargs(supplied_config, &count);
+            if (count != 2) {
+                serverLog(LL_WARNING, "Couldn't load configuration: malformed arguments");
+                sdsfreesplitres(config_pair, count);
+                sdsfree(config_name);
+                listEmpty(server.module_configs_queue);
+                return REDISMODULE_ERR;
+            }
+            if (!(strcasecmp(config_name, config_pair[0]))) {
+                if (!performConfigSetFromName(config_pair[0], config_pair[1], &err)) {
+                    serverLog(LL_WARNING, "Issue during loading of configuration %s : %s", config_pair[0], err);
+                    sdsfreesplitres(config_pair, count);
+                    sdsfree(config_name);
+                    listEmpty(server.module_configs_queue);
+                    return REDISMODULE_ERR;
+                }
+                config_supplied = 1;
+                sdsfreesplitres(config_pair, count);
+                listDelNode(server.module_configs_queue, ln_supplied);
+                break;
+            }
+            sdsfreesplitres(config_pair, count);
+        }
+        
+        if (!config_supplied) {
+            if (!performModuleConfigInitFromName(config_name, &err)) {
+                serverLog(LL_WARNING, "Issue attempting to set default value of configuration %s : %s", module_config->name, err);
+                listEmpty(server.module_configs_queue);
+                sdsfree(config_name);
+                return REDISMODULE_ERR;
+            }
+        }
+        sdsfree(config_name);
+        config_supplied = 0;
+    }
+    return REDISMODULE_OK;
+}
+
+/* Add module_config to the list if the apply and privdata do not match one already in it. */
+void addModuleApply(list *module_configs, ModuleConfig *module_config) {
+    if (!module_config->apply_fn) return;
+    listIter li;
+    listNode *ln;
+    ModuleConfig *pending_apply;
+    listRewind(module_configs, &li);
+    while ((ln = listNext(&li))) {
+        pending_apply = ln->value;
```

Comment:
```suggestion
        pending_apply = listNodeValue(ln);
```
I know it's not widely done it redis, but we should try to respect APIs.